src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIBookingService.kt

```diff
@@ -51,26 +75,42 @@ class CommunityAPIBookingService(
       notes = resourceUrl,
       countsTowardsRarDays = true, // Fixme: For assessment booking this should be false and will pass in when assessment booking is done
     )
+  }
 
-    val referral = appointment.actionPlan.referral
-    val communityApiBookAppointmentPath = UriComponentsBuilder.fromPath(communityApiBookAppointmentLocation)
-      .buildAndExpand(referral.serviceUserCRN, referral.relevantSentenceId!!, integrationContext)
-      .toString()
+  private fun buildAppointmentRescheduleRequestDTO(appointmentTime: OffsetDateTime, durationInMinutes: Int): AppointmentRescheduleRequestDTO {
 
-    val response = communityAPIClient.makeSyncPostRequest(communityApiBookAppointmentPath, appointmentCreateRequestDTO, AppointmentCreateResponseDTO::class.java)
-    logger.debug("Requested booking for appointment. Returned appointment id: $response")
+    return AppointmentRescheduleRequestDTO(
+      updatedAppointmentStart = appointmentTime,
+      updatedAppointmentEnd = appointmentTime.plusMinutes(durationInMinutes.toLong()),
+      initiatedByServiceProvider = true, // fixme - needs to come from the user - defaulted to SP Initiated Reschedule
+    )
+  }
 
-    return response.appointmentId
+  private fun buildReferralResourceUrl(existingAppointment: ActionPlanAppointment): String {
+    return UriComponentsBuilder.fromHttpUrl(interventionsUIBaseURL)
+      .path(interventionsUIViewAppointment)
+      .buildAndExpand(existingAppointment.actionPlan.referral.id)
+      .toString()
   }
 
   fun isInitialBooking(existingAppointment: ActionPlanAppointment, appointmentTime: OffsetDateTime?, durationInMinutes: Int?): Boolean =
     isTimingSpecified(appointmentTime, durationInMinutes) &&
       !isTimingSpecified(existingAppointment.appointmentTime, existingAppointment.durationInMinutes)
 
+  fun isRescheduleBooking(existingAppointment: ActionPlanAppointment, appointmentTime: OffsetDateTime?, durationInMinutes: Int?): Boolean =
```

Comment:
so this is distinct from `!isInitialBooking` because the timing is different. what happens if the user just selects the same time as the previous appointment? i guess nothing? is 200 still a suitable response in this case?

Reply:
isReschduleBooking also checks isDifferentTimings. If the timings have not changed then the reschedule booking is not activated and an appointmentId not returned to the caller, meaning no action. The deliusAppointmentId remains unchanged.

Reply:
Will add test to demonstrate it.